python/ray/data/dataset.py

```diff
@@ -280,6 +280,26 @@ def map(
                 must be used.
             ray_remote_args: Additional resource requirements to request from
                 ray (e.g., num_gpus=1 to request GPUs for the map tasks).
+
+        .. seealso::
```

Comment:
Why this seealso is only for .map(), but not other mapping apis? Also .map() itself is linked as seealso, is that intended?

Reply:
> Why this seealso is only for .map(), but not other mapping apis? Also .map() itself is linked as seealso, is that intended?

No, this isn't intended for the final PR. Sorry, forgot to mark this as a draft.

Was gonna get feedback on the three descriptions before adding it to all of the map methods.

Reply:
An idea proposal: having a table to compare all 3 map apis and provide decision tree for users to help them choose which one to use

Reply:
IMO I agree with @Wendi-anyscale. Could we add more information into documentation https://docs.ray.io/en/latest/data/transforming-datasets.html#transformations instead?

We can add more information to compare `map`, `flat_map`, `map_batches` there, currently it does not provide much information:
<img width="958" alt="Screen Shot 2022-11-28 at 9 49 32 AM" src="https://user-images.githubusercontent.com/4629931/204346341-6264ae5b-52b6-43ed-b75f-f54ec69e9261.png">

Reply:
> IMO I agree with @Wendi-anyscale. Could we add more information into documentation https://docs.ray.io/en/latest/data/transforming-datasets.html#transformations instead?

Would it make sense to add both? I can add changes to the user guide in a separate PR.

Reply:
Discussed with @bveeramani offline. It will be useful to add the `seealso` here, and also make the existing summary table more discoverable. Other system like [`PySpark` also has `seealso` section](https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.Series.map.html#pyspark.pandas.Series.map).

I will change the layout of our API reference, to make the existing summary table more discoverable.

Reply:
Discussed with @bveeramani offline. It will be useful to add the `seealso` here, and also make the existing summary table more discoverable. Other system like [`PySpark` also has `seealso` section](https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.Series.map.html#pyspark.pandas.Series.map).

I will change the layout of our API reference, to make the existing summary table more discoverable.

Reply:
Discussed with @bveeramani offline. It will be useful to add the `seealso` here, and also make the existing summary table more discoverable. Other system like [`PySpark` also has `seealso` section](https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.Series.map.html#pyspark.pandas.Series.map).

I will change the layout of our API reference, to make the existing summary table more discoverable.

Reply:
@jianoaix - please take a look of detailed plan where it makes sense - https://github.com/ray-project/ray/issues/30692 .